core/domain/user_query_jobs_one_off_test.py

```diff
@@ -0,0 +1,157 @@
+# coding: utf-8
+#
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for query MR job."""
+
+from core.domain import exp_services
+from core.domain import user_query_jobs_one_off
+from core.domain import user_query_services
+from core.platform import models
+from core.tests import test_utils
+
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+taskqueue_services = models.Registry.import_taskqueue_services()
+
+
+class UserQueryJobOneOffTests(test_utils.GenericTestBase):
+    EXP_ID_1 = 'exp_id_1'
+    EXP_ID_2 = 'exp_id_2'
+    USER_A_EMAIL = 'a@example.com'
+    USER_A_USERNAME = 'a'
+    USER_B_EMAIL = 'b@example.com'
+    USER_B_USERNAME = 'b'
+    USER_C_EMAIL = 'c@example.com'
+    USER_C_USERNAME = 'c'
+    USER_D_EMAIL = 'd@example.com'
+    USER_D_USERNAME = 'd'
+    USER_SUBMITTER_EMAIL = 'submit@example.com'
+    USER_SUBMITTER_USERNAME = 'submit'
+
+    def _run_one_off_job(self, query_id):
+        """Runs the one-off MapReduce job."""
+        job_id = user_query_jobs_one_off.UserQueryOneOffJob.create_new()
+        params = {
+            'query_id': query_id
+        }
+        user_query_jobs_one_off.UserQueryOneOffJob.enqueue(
+            job_id, additional_job_params=params)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                queue_name=taskqueue_services.QUEUE_NAME_DEFAULT), 1)
+        self.process_and_flush_pending_tasks()
+
+    def setUp(self):
```

Comment:
Where's the test for "recently logged in"? This seems to have been omitted from the job, too.